File: src/main/java/Practicas/Practica2/Fusioner.java
package Practicas.Practica2;
import java.io.*;
import java.util.*;

public class Fusioner {
    public static void main(String[] args) {
        Map<String, Integer> totalFreq = new HashMap<>();
        for (String archivo : args) {
            try (BufferedReader br = new BufferedReader(new FileReader(archivo))) {
                String linea;
                while ((linea = br.readLine()) != null) {
                    String[] partes = linea.split(",");
                    if (partes.length == 2) {
                        String palabra = partes[0];
                        int frecuencia = Integer.parseInt(partes[1]);
                        totalFreq.put(palabra, totalFreq.getOrDefault(palabra, 0) + frecuencia);
                    }
                }
            } catch (IOException e) {
                e.printStackTrace();
            }
        }

        System.out.println("5 palabras más frecuentes:");
        totalFreq.entrySet().stream()
                .sorted((a, b) -> b.getValue().compareTo(a.getValue()))
                .limit(5)
                .forEach(e -> System.out.println(e.getKey() + ": " + e.getValue()));

        for (String archivo : args) {
            new File(archivo).delete();
        }
    }
}

File: untitled/PythonFusioner/Fusioner.py
import os

class Fusioner:
    @staticmethod
    def main(args):
        total_freq = dict(int)
        for archivo in args:
            try:
                with open(archivo) as a:
                    for linea in a:
                        partes = partes.split[","]
                        if len(partes)==2:
                            palabra = partes[0]
                            frecuencia = int(partes[1])
                            total_freq[palabra] += frecuencia
                        pass
            except IOError:
                print("Error de lectura")

        print("5 palabras más frecuentes")
        top5 = sorted(total_freq.items(), key=lambda x: x[1], reverse=True)[:5]

        for palabra, frecuencia in top5:
            print(f"{palabra}: {frecuencia}")

        for archivo in args:
            try:
                os.remove(archivo)
            except OSError:
                print("El archivo no existe.")
    

            
